extensions/mongoProcessor/MongoQueueProcessor.js

```diff
@@ -210,9 +231,39 @@ class MongoQueueProcessor {
             isNFS: null,
         };
 
+        if (role && destination && rules && rules.length > 0) {
```

Comment:
Could be worthwhile to abstract and reuse the logic in https://github.com/scality/cloudserver/blob/development/8.1/lib/api/apiUtils/object/getReplicationInfo.js#L95.

Reply:
In the interest of time, I'll be duplicating some code here rather than reusing the cloudserver code. I'll follow up to refactor the code